server/game/gameSystems/ReadyResourcesSystem.ts

```diff
@@ -0,0 +1,41 @@
+import { AbilityContext } from '../core/ability/AbilityContext';
+import type { Card } from '../core/card/Card';
+import { CardType, Location, WildcardCardType } from '../core/Constants';
+import Player from '../core/Player.js';
+import * as EnumHelpers from '../core/utils/EnumHelpers';
+import { IPlayerTargetSystemProperties, PlayerTargetSystem } from '../core/gameSystem/PlayerTargetSystem.js';
+
+
+export interface IReadyResourcesSystemProperties extends IPlayerTargetSystemProperties {
+    amount?: number
+}
+
+export class ReadyResourcesSystem<TContext extends AbilityContext = AbilityContext, TProperties extends IReadyResourcesSystemProperties = IReadyResourcesSystemProperties> extends PlayerTargetSystem<TContext, TProperties> {
+    protected override defaultProperties: IReadyResourcesSystemProperties = {
+        amount: 1
+    };
+
```

Comment:
Need to set an eventName property so that this system's event could be responded to with a triggered ability.

We have onSpendResources but adding onReadyResources and onExhaustResources would probably be clearer